athena-msk/src/main/java/com/amazonaws/athena/connectors/msk/AmazonMskRecordHandler.java

```diff
@@ -80,35 +87,90 @@ public void readWithConstraint(BlockSpiller spiller, ReadRecordsRequest recordsR
         // Taking the Split parameters in a readable pojo format.
         SplitParameters splitParameters = AmazonMskUtils.createSplitParam(recordsRequest.getSplit().getProperties());
         LOGGER.info("[kafka] {} RecordHandler running", splitParameters);
+        // Set which topic and partition we are going to read.
+        TopicPartition partition = new TopicPartition(splitParameters.topic, splitParameters.partition);
+        Collection<TopicPartition> partitions = com.google.common.collect.ImmutableList.of(partition);
+        GlueRegistryReader registryReader = new GlueRegistryReader();
 
-        // Initiate new KafkaConsumer that MUST not belong to any consumer group.
-        try (Consumer<String, TopicResultSet> kafkaConsumer = AmazonMskUtils.getKafkaConsumer(recordsRequest.getSchema(), configOptions)) {
-            // Set which topic and partition we are going to read.
-            TopicPartition partition = new TopicPartition(splitParameters.topic, splitParameters.partition);
-            Collection<TopicPartition> partitions = com.google.common.collect.ImmutableList.of(partition);
+        String dataFormat = registryReader.getGlueSchemaType(recordsRequest.getTableName().getSchemaName(), recordsRequest.getTableName().getTableName());
```

Comment:
this method is getting really long; we have now three conditions; can we adopt a better pattern; split out the smaller logic into their own functions; if it is possible; extract common code to be used across all three conditions.

Reply:
Moreover; I believe we can adopt an [strategy pattern](https://www.baeldung.com/java-strategy-pattern) here and even extract all the `xxxConsume` & `xxxExecute` outside of this class to make it easier to read; most important maintain.

If we also have to introduce a new type we can also do this much easier.

Reply:
Hi @AbdulR3hman , i have moved these consumers to other package.

Reply:
this is so much cleaner; easier to read and reason about; thank you.